### PR TITLE
DS-2895 authorization check for item not owned by collection

### PR DESCRIPTION
**Pull request by abollini**
I don't like the cyclic dependency between ItemService, WorkspaceItem and WorkflowItem but I don't see a different way to fix the issue. The ItemService should be not responsible to implement authorization logic as in the canEdit method.
We could move the canEdit method in the authorization service but this will lead anyway to a cyclic reference (ItemService -> AuthorizationService -> ItemService / WorkspaceItemService/ WorkflowItemService).

Anyway, the cyclic dependency is not an issue, it is managed properly by Spring it is just a not nice thing that show a bad SoC

### CODE REVIEW

**Comment by abollini** [comment]
I have fixed the issue found by the WorkspaceItemTest (creation of new submission) related to an authorization exception throw by the updateLastModified call during workspaceitem creation.
To alter the resourcepolicy it was not actually necessary to have WRITE permission, it is possible only trough API and the security need to be enforced by the UI for example using a functional logic (if you can create an object you can decide the policy for such object, etc.).
Please note that I have also moved the updateLastModified call after the code that perform the resource policy change to be more consistent (before addPolicy and delete* work in a different order).

File: dspace-api/src/main/java/org/dspace/content/ItemServiceImpl.java [hunk 1]
```diff
@@ -881,12 +887,28 @@ public boolean canEdit(Context context, Item item) throws SQLException {
         // is this collection not yet created, and an item template is created
         if (item.getOwningCollection() == null)
         {
-            return true;
+        	if (!isAnInprogressSubmission(context, item)) {
```
Comment (mwoodiupui) [inline dspace-api/src/main/java/org/dspace/content/ItemServiceImpl.java]: Why not just 'return !isAnInprogressSubmission(context, item);'?

File: dspace-api/src/main/java/org/dspace/content/ItemServiceImpl.java [hunk 2]
```diff
@@ -78,6 +79,11 @@
     protected VersioningService versioningService;
     @Autowired(required=true)
     protected HarvestedItemService harvestedItemService;
+    @Autowired(required=true)
+    protected WorkspaceItemService workspaceItemService;
```
Comment (mwoodiupui) [inline dspace-api/src/main/java/org/dspace/content/ItemServiceImpl.java]: I don't see an import for this.

**Comment by tdonohue** [comment]
+1 via code review.

File: dspace-api/src/main/java/org/dspace/content/ItemServiceImpl.java (hunk 2 above)
Comment (abollini) [inline dspace-api/src/main/java/org/dspace/content/ItemServiceImpl.java]: we have imported the full .service package, see line 21

File: dspace-api/src/main/java/org/dspace/content/ItemServiceImpl.java (hunk 1 above)
Comment (abollini) [inline dspace-api/src/main/java/org/dspace/content/ItemServiceImpl.java]: probably we should improve this logic in future...

**Comment by abollini** [comment]
ready to merge

**Comment by mwoodiupui** [comment]
+1 by inspection

**Comment by tdonohue** [comment]
Unit tests provide great coverage to prove out this works right.  Approved in today's DevMtg. Merging.

**Pull request merged**